src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ConfiguredApiRepositoryCustom.kt

```diff
@@ -0,0 +1,74 @@
+package uk.gov.justice.digital.hmpps.digitalprisonreportingmi.data
+
+import jakarta.validation.ValidationException
+import org.apache.commons.lang3.time.StopWatch
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Service
+import java.sql.Timestamp
+
+@Service
+@Suppress("UNCHECKED_CAST")
+class ConfiguredApiRepositoryCustom {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Autowired
+  lateinit var jdbcTemplate: NamedParameterJdbcTemplate
+  fun executeQuery(
+    query: String,
+    rangeFilters: Map<String, String>,
+    filtersExcludingRange: Map<String, String>,
+    selectedPage: Long,
+    pageSize: Long,
+    sortColumn: String,
+    sortedAsc: Boolean,
+  ): List<Map<String, Any>> {
+    val preparedStatementNamedParams = MapSqlParameterSource()
+    // TODO: Consider case insensitive non range filters
```

Comment:
Probably prudent, given the inconsistent casing in some of the data.

Reply:
Done.